src/Microsoft.ML.Transforms/PermutationFeatureImportance.cs

```diff
@@ -139,7 +139,7 @@ internal static class PermutationFeatureImportance<TMetric, TResult> where TResu
                 var valuesRowCount = 0;
                 // REVIEW: Seems like if the labels are NaN, so that all metrics are NaN, this command will be useless.
                 // In which case probably erroring out is probably the most useful thing.
-                using (var cursor = data.GetRowCursor(col => col == featuresColumnIndex))
+                using (var cursor = data.GetRowCursor(data.Schema.Where(x =>  x.Index == featuresColumnIndex)))
```

Comment:
>data.Schema.Where(x =>  x.Index == featuresColumnIndex [](start = 54, length = 54)

Much better than this of course would be `new[] { data.Schema[featuresColumnIndex] }`. :) But I would actually suggest actually something else, I suspect that it would be best actually to have an extension method to `IDataView` somewhere in our public API with the signature `RowCursor GetRowCursor(params Schema.Column[] columns)` to handle this case, which I must expect would be very common.

Reply:
I wasn't 100% sure that x.Index and the position of the columns in the Shema matches, hence this checking on the Index everywhere. 

After I get all the tests passing, I'll swap this out everywhere.

---
In reply to: [244382663](https://github.com/dotnet/machinelearning/pull/1975#discussion_r244382663) [](ancestors = 244382663)

Reply:
OK! Also w.r.t. the extension, if you see that the single column selection is common, might even be good to have extension overloads for `Schema.Column`, and `Schema.Column, Schema.Column` (for two accessors, again if you observe it being common), in addition to the suggested `params` extension.

---
In reply to: [244384709](https://github.com/dotnet/machinelearning/pull/1975#discussion_r244384709) [](ancestors = 244384709,244382663)

Reply:
This was marked as resolved, could you tell me @sfilipi if you added these extensions anywhere? The closest thing I saw that is new since my last review was `DataViewExtensions`, but that doesn't operate over `Schema.Column` as it should, but over `string` and `int`, which is very dangerous.

---
In reply to: [244385783](https://github.com/dotnet/machinelearning/pull/1975#discussion_r244385783) [](ancestors = 244385783,244384709,244382663)

Reply:
Added now, will show up in the next PR update soon. 

---
In reply to: [245083396](https://github.com/dotnet/machinelearning/pull/1975#discussion_r245083396) [](ancestors = 245083396,244385783,244384709,244382663)